.github/workflows/gh_release.yml

```diff
@@ -0,0 +1,59 @@
+# This workflow will install Python dependencies, run pre-release script, and
+# then bundle a github release
+
+name: Release generator
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  context:
+    if: github.base_ref == 'master' && github.event.pull_request.merged && contains( github.event.pull_request.labels.*.name, 'release')
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Create tag from title and run asset script
+        id: find_tag_and_prepare_assets
+        run: |
+          TAG=$(echo ${{ github.event.pull_request.title }} | sed -E "s/^.*Release (.+\..+\..+)$/\1/g")
```

Comment:
The PR titles _should_ all be simpler than the full range of valid semver. But we can try it!

Reply:
The official regex is pretty complex. As written, it breaks sed on macos. :\

Reply:
Alright, which do you prefer:

```
# https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
SEMVER="(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
TAG=$(echo ${{ github.event.pull_request.title }} | perl -pe "s/^.*Release ${SEMVER}\$/\1.\2.\3/g")
```
or
```
TAG=$(echo ${{ github.event.pull_request.title }} | sed -E "s/^.*Release (.+\..+\..+)$/\1/g")
```
?

Reply:
The only other thing I can think of that doesn't require a specific PR title format would be to have the CLI create a second label `#.#.#` which this workflow step could use and then delete after its finished? 

This might also come in handy for the user provided script in step 8 of `What the CLI Does`. I'm thinking for most of the repos with Python packages, the user provided script will be used to update the package's `__init__.py`  with the current version. For the `kf-model-fhir` repo, the user script will also need the version to in order to tag the FHIR model's resources with it.

Reply:
> the user script will also need the version

Done in https://github.com/d3b-center/d3b-release-maker/pull/1/commits/f4b211927d970aa444f9968ecfb30eb365c4d10d

> create a second label #.#.# which this workflow step could use and then delete after its finished

Ooh, I like that. I'll see if it works and post that change if it does.

Reply:
> Done in f4b2119

I think we're talking about two different user scripts. You're referring to the prepare assets script that gets run after the release and I'm referring to the pre-release script that gets executed in step 8 of `release build` right before the release branch is pushed up to Github.

Although, now that you mention the environment variable.. Maybe the CLI could write the version to a environment variable (e.g. `GH_RELEASE_VERSION`, something more specific than `TAG`) and then the pre-release script can read the version from there. 

If we can do this, then I'd say let's keep this workflow step (get tag) simple and stick with your first regex to get the version from the PR title?

Reply:
https://github.com/d3b-center/d3b-release-maker/pull/1/commits/a32a0c9c117e9c9d12d6995caf4640627ca27fb0

Reply:
the script is passed the version as input. if it wants to set environment variables it can.